packages/compass-indexes/src/modules/regular-indexes.ts

```diff
@@ -541,20 +539,18 @@ export const dropIndex = (
   ) => {
     const { namespace, regularIndexes } = getState();
     const { indexes, inProgressIndexes } = regularIndexes;
-    const index = indexes.find((x) => x.name === indexName);
-
-    if (!index) {
-      return;
-    }
 
+    // TODO: this should be its own function, not part of dropIndex
```

Comment:
There was actually momentarily a bug here. Because I'm not merging the in-progress indexes into indexes, the code:

```
const index = indexes.find((x) => x.name === indexName);

    if (!index) {
      return;
    }
```

which ran first meant that you could never get to the check to see if the thing is an in-progress index. Which wasn't as obvious to find and I think if the two things weren't both rolling into dropIndex() then it would have been easier to spot, more likely to be tested, etc.